### ISSUE TITLE
Abstract classes that implement Monobehaviour routines should allow for inheritance via protected virtual access modifiers.

### ISSUE THREAD

**Issue by ahokinson** [mapbox/mapbox-unity-sdk#647]
I've made this point before in other issues I've submitted and I just updated to 1.4.0 where the issue persists.

If AbstractMap.cs is intended to be abstract and built upon further to create specialized maps, the use of Awake, Start, and other MonoBehaviour methods may need to be modified.

In my specific case, I would like to use Awake() and Start() in my inherited class, but am blocked from doing so by the inexplicit private modifiers on those methods.

```
void Awake()...
void Start()...
```

should be replaced with

```
protected virtual void Awake()...
protected virtual void Start()...
```

Because both of these are blocked, I am unable to run a related method without modifying the Mapbox source.

e.g.
```
protected override void Start()
{            
    base.Start();
    Root.parent.GetComponentsInChildren<SphereCapper>().ToList().ForEach(sc => sc.GenerateCapMesh());
}
```

I am willing to do a pull request for this change, but I understand that access modifiers should be left to Mapbox's discretion.

**Comment by atripathi-mb** [comment]
@ahokinson Please create a pull request and we can evaluate how best to allow access to such requirements.

Also, would calling your method after `OnInitialized` event is triggered be a viable solution?

**Comment by ahokinson** [comment]
If AbstractMap.cs inherits from MonoBehaviour, I don't feel like further use of MonoBehaviour methods for inherited classes should be restricted by private modifiers. So while I'm sure that I could get OnInitialized to work for me, it doesn't change the fact that inherited classes may have valid uses for MonoBehaviour methods.

I'll put together a PR.

**Comment by atripathi-mb** [comment]
@ahokinson 
> As an aside, if a method or member is intended to beprivate, shouldn't it be marked explicitly to show intent even if they mean the same thing?

Yes, will add this to our coding /contribution guides for future releases. Thanks for pointing this out.
